geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/LuceneServiceImpl.java

```diff
@@ -221,8 +222,12 @@ private void createIndexOnExistingRegion(PartitionedRegion region, String indexN
       LuceneSerializer serializer) {
     validateRegionAttributes(region.getAttributes());
 
-    region.addCacheServiceProfile(new LuceneIndexCreationProfile(indexName, regionPath, fields,
-        analyzer, fieldAnalyzers, serializer));
+    try {
+      region.addCacheServiceProfile(new LuceneIndexCreationProfile(indexName, regionPath, fields,
+          analyzer, fieldAnalyzers, serializer));
+    } catch (IncompatibleCacheServiceProfileException e) {
+      e.printStackTrace();
```

Comment:
In my opinion we need to clean up the cacheProfile that we added in the addCacheServiceProfile if an exception occurred and prevent the create index execution to continue